### PR TITLE
Clarify children access in renderEditor in docs

### PR DESCRIPTION
**Pull request by klis87**
#### Is this adding or improving a _feature_ or fixing a _bug_?
Docs improvement related to #2259 

#### Have you checked that...?

* [ ] The new code matches the existing patterns and styles.
* [ ] The tests pass with `yarn test`.
* [ ] The linter passes with `yarn lint`. (Fix errors with `yarn prettier`.)
* [ ] The relevant examples still work. (Run examples with `yarn watch`.)

#### Does this fix any issues or need any specific reviewers?

Fixes: #
Reviewers: @

### CODE REVIEW

File: docs/reference/slate-react/plugins.md [hunk 1]
```diff
@@ -109,7 +109,15 @@ This handler is called whenever the native DOM selection changes.
 
 `Function renderEditor(props: Object, next: Function) => ReactNode|Void`
 
-The `renderEditor` property allows you to define higher-order-component-like behavior. It is passed all of the properties of the editor, including `props.children`. You can then choose to wrap the existing `children` in any custom elements or proxy the properties however you choose. This can be useful for rendering toolbars, styling the editor, rendering validation, etc. Remember that the `renderEditor` function has to render `props.children` for editor's content to render.
+The `renderEditor` property allows you to define higher-order-component-like behavior. It is passed all of the properties of the editor, including `children`, which you can access as `next()`. You can then choose to wrap the existing `children` in any custom elements or proxy the properties however you choose. This can be useful for rendering toolbars, styling the editor, rendering validation, etc. Remember that the `renderEditor` function has to render `children` for editor's content to render. For example:
```
Comment (ianstormtaylor) [inline docs/reference/slate-react/plugins.md]: Can you explain here that `next()` retrieves the children by deferring to the other plugins in the stack?

File: docs/reference/slate-react/plugins.md [hunk 2]
```diff
@@ -109,7 +109,15 @@ This handler is called whenever the native DOM selection changes.
...
+```js
+renderEditor: ({ editor }, next) => (
```
Comment (ianstormtaylor) [inline docs/reference/slate-react/plugins.md]: Can you change this to be `(props, next)` without the destructuring to not confuse people?

File: docs/reference/slate-react/plugins.md [hunk 3]
```diff
@@ -109,7 +109,15 @@ This handler is called whenever the native DOM selection changes.
...
+  <div>
+    <MyToolbarComponent editor={editor} />
+    <MyEditorComponent editor={editor}>{next()}</MyEditorComponent>
```
Comment (ianstormtaylor) [inline docs/reference/slate-react/plugins.md]: Can you set this as a variable `const children = next()` to make it more explicit for people?

File: docs/reference/slate-react/plugins.md (hunk 2 above)
Comment (valorize) [reply docs/reference/slate-react/plugins.md]: As far as I see in https://github.com/ianstormtaylor/slate/blob/master/examples/plugins/word-count.js#L14 the second parameter is editor and the thrid is next
```suggestion
renderEditor: (props, editor, next) => (
```

Comment (klis87) [reply docs/reference/slate-react/plugins.md]: @valorize probably it has changed recently, during writing this PR next was passed as 2nd argument

Comment (klis87) [reply docs/reference/slate-react/plugins.md]: @valoricDe @ianstormtaylor I updated PR to match to (props, editor, next) function signature

Comment (valoricDe) [reply docs/reference/slate-react/plugins.md]: @klis87 👍 But sadly the build step is failing. Could you check what the output of `prettier  '**/*.{md,json,css}'` for the file `docs/reference/slate-react/plugins.md` is?

Comment (klis87) [reply docs/reference/slate-react/plugins.md]: @valoricDe I wasn't aware prettier is used for md files, I pushed the prettified version and now build passes

**Comment by ianstormtaylor** [comment]
Thanks @klis87!

**Pull request merged**